youtube_dl/extractor/twitch.py

```diff
@@ -77,15 +78,19 @@ def login_step(page, urlh, note, data):
             page_url = urlh.geturl()
             post_url = self._search_regex(
                 r'<form[^>]+action=(["\'])(?P<url>.+?)\1', page,
-                'post url', default=page_url, group='url')
+                'post url', default=self._LOGIN_POST_URL, group='url')
             post_url = urljoin(page_url, post_url)
 
-            headers = {'Referer': page_url}
+            headers = {
+                'Referer': page_url,
+                'Origin': page_url,
+                'Content-Type': 'text/plain;charset=UTF-8'
+            }
 
             try:
                 response = self._download_json(
                     post_url, None, note,
-                    data=urlencode_postdata(form),
+                    data=json.dumps(form),
                     headers=headers)
             except ExtractorError as e:
                 if isinstance(e.cause, compat_HTTPError) and e.cause.code == 400:
```

Comment:
Error handling broken.

Reply:
Is this related to the way I incorrectly handled the encoding above or something else? If something else how can I test it?

Reply:
JSON layout has changed. Error is not captured properly anymore.